pkg/network/protocols/events/configuration.go

```diff
@@ -0,0 +1,62 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package events
+
+import (
+	"os"
+	"sync"
+
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+	"github.com/iovisor/gobpf/pkg/cpupossible"
+)
+
+var handlerMux sync.Mutex
+var handlerByProtocol map[string]*ddebpf.PerfHandler
+
+// Configure event processing
+// Must be called *before* manager.InitWithOptions
+func Configure(proto string, m *manager.Manager, o *manager.Options) {
```

Comment:
can it be called in parallel? if yes, then why? if no, then I think we can remove the mutex, and move the initialization of the map into `def init`

Reply:
The answer is _maybe_ - it will depend on how the package is used.
IMO there is no gain in removing the mutex and having a potential race condition